### ISSUE TITLE
question: status of the `dev` branch

### ISSUE THREAD

**Issue by WieeRd** [barreiroleo/ltex_extra.nvim#60]
I've noticed the `dev` branch going on and switched to test it - around 2 months ago. So far, it has worked flawlessly. The LspAttach based automatic setup really did make my config cleaner as well, compared to manual setup using `on_attach()`.

Today I rechecked this repo and was surprised that it still hasn't been merged yet. Both the master and dev branch have been quite for a while, so I assume you are busy with other things.

Are there unresolved problems / unimplemented features in `dev` branch that's stopping it from being merged? Whatever it is, I am willing to help. I really appreciate this plugin - I needed that 'add to dictionary' support :)

**Comment by barreiroleo** [comment]
Hi hi!
The main blocker to doing the merge it's the feedback. Giving the nature of this plugin the last thing that I want is to release a breaking change with knock-ons. So, glad to hear dev branch is working without problems.
I think I can push a tagged version for master though.

**Comment by WieeRd** [comment]
Isn't current `dev` branch already quite backwards compatible? The ideal way to introduce breaking change is to

1. Keep the legacy config fully functional for a while.
2. Notice the user with deprecation warnings when the use of legacy features is detected.

And looks like you are already doing both:

https://github.com/barreiroleo/ltex_extra.nvim/blob/57192d7ae5ba8cef3c10e90f2cd62d4a7cdaab69/lua/ltex_extra/init.lua#L168-L171

https://github.com/barreiroleo/ltex_extra.nvim/blob/57192d7ae5ba8cef3c10e90f2cd62d4a7cdaab69/lua/ltex_extra/init.lua#L136-L147

If you can just tag the current `master` as `v0.3` and `dev` as `v1.0` after merging, I think it's good to go!

**Comment by WieeRd** [comment]
Perhaps it might be worthwhile to add a check / deprecation notice for the 2nd invocation of `require("ltex_extra").setup()`, since it strongly indicates that the setup is being called inside `on_attach()` as a legacy method of setting up this plugin, and being called every time a new markdown buffer is opened.

**Comment by barreiroleo** [comment]
Yeah, it's supposed to be backwards compatible as it is, but you know.. each nvim config is a different world.
I saw the PR, many thanks. I'll give it a try and then merge it as soon as posible. Busy days

**Comment by WieeRd** [comment]
These days, with other Neovim plugins making breaking changes like every other week, I appreciate your efforts to keep the code compatible and minimize the friction for the users. Wish you good luck with whatever irl stuffs you're working on :)

**Comment by 89iuv** [comment]
I am on the dev branch and it works without a problem for me. 
If there is anything i can do to help please let me know.

This plugin saved me a lot of time and headaches.

**Comment by WieeRd** [comment]
@barreiroleo Happy new year! Just a small reminder ping since 3 months have passed since.

**Comment by barreiroleo** [comment]
Hi all, happy holidays and happy new year!

Sorry for missing the thread, last part of the year was a bit crazy.
I will try to give the #61 a shot over the weekend, push a new tag for the current version and move on.

Ty for the reminder btw

**Comment by barreiroleo** [comment]
Well, I took a look at this and I needed to implement some test files.

Need to test on nvim 0.9 before merge; `dev` will become the new main and I already branched the current main in `legacy` branch.

**Comment by NicolaiRuckel** [comment]
Are there any updates on this?

**Comment by barreiroleo** [comment]
https://github.com/barreiroleo/ltex_extra.nvim/issues/75
